TipCatDotNet.Api/Services/Payments/TransactionService.cs

```diff
@@ -94,6 +101,32 @@ async Task<Result> SetPaymentTime(int memberId)
     }
 
 
+    public async Task<Result<List<FacilityTransactionResponse>>> Get(MemberContext context,
+        TransactionFilterProperty filterProperty, CancellationToken cancellationToken = default)
+    {
+        var facilities = await _context.Facilities
+            .Where(f => f.AccountId == context.AccountId)
+            .ToListAsync(cancellationToken);
+
+        var transactions = _context.Transactions
+            .Select(TransactionProjection())
+            .AsEnumerable()
+            .GroupBy(t => t.FacilityId)
+            .ToDictionary(t => t.Key, t => t.ToList());
+
+
+        var result = new List<FacilityTransactionResponse>(facilities.Count);
+        foreach (var facility in facilities)
+        {
+            transactions.TryGetValue(facility.Id, out var facilityTransactions);
+            var total = (facilityTransactions != null) ? facilityTransactions.Sum(t => t.Amount.Amount) : 0;
+            result.Add(new FacilityTransactionResponse(facility.Id, facility.Name, total, facilityTransactions));
```

Comment:
Not quite understand what you try to achieve here? You may use something like [this](https://stackoverflow.com/questions/50162978/ef-core-group-by-translation-support-on-conditional-sum) to aggregate. Also, I don't think you need all the amount, but within a limited time-frame, a month for instance

Reply:
I did it like [this](https://github.com/tipcat-net/tipcat-net-api/blob/master/TipCatDotNet.Api/Services/HospitalityFacilities/FacilityService.cs#:~:text=var%20result%20%3D%20new,%7D)